drivers/gpio/gpio_stmpe1600.c

```diff
@@ -0,0 +1,324 @@
+			       gpio_pin_t pin, gpio_flags_t flags)
+{
+	int ret;
+	struct stmpe1600_drvdata *const drvdata = (struct stmpe1600_drvdata *const) dev->data;
+	const struct stmpe1600_config *const config = dev->config;
+
+	/* No support for disconnected pin */
+	if ((flags & (GPIO_INPUT | GPIO_OUTPUT)) == GPIO_DISCONNECTED) {
+		return -ENOTSUP;
+	}
+
+	/* STMPE1600 does not support any of these modes */
+	if ((flags & (GPIO_SINGLE_ENDED | GPIO_PULL_UP | GPIO_PULL_DOWN)) != 0) {
+		return -ENOTSUP;
+	}
+
+	if (k_is_in_isr()) {
+		return -EWOULDBLOCK;
+	}
+
+	k_sem_take(&drvdata->lock, K_FOREVER);
+
+	ret = set_pin_dir(dev, pin, flags);
+	if (ret != 0) {
+		LOG_ERR("STMPE1600[0x%X]: error setting pin direction (%d)",
+			config->i2c_slave_addr, ret);
+	}
+
+	k_sem_give(&drvdata->lock);
+	return ret;
+}
+
+static int stmpe1600_port_get_raw(const struct device *dev, uint32_t *value)
+{
+	int ret;
+	uint16_t reg_value;
+	struct stmpe1600_drvdata *const drvdata = (struct stmpe1600_drvdata *const) dev->data;
+
+	if (k_is_in_isr()) {
+		return -EWOULDBLOCK;
+	}
+
+	k_sem_take(&drvdata->lock, K_FOREVER);
+	ret = read_reg16(dev->config, REG_GPMR_LSB, &reg_value);
+	k_sem_give(&drvdata->lock);
+
+	if (ret == 0) {
+		*value = reg_value;
+	}
+
+	return ret;
+}
+
+static int stmpe1600_port_set_masked_raw(const struct device *dev,
+					 uint32_t mask, uint32_t value)
+{
+	int ret;
+	uint16_t GPSR;
+	struct stmpe1600_drvdata *const drvdata = (struct stmpe1600_drvdata *const)dev->data;
+
+	if (k_is_in_isr()) {
+		return -EWOULDBLOCK;
+	}
+
+	k_sem_take(&drvdata->lock, K_FOREVER);
+	GPSR = (drvdata->GPSR & ~mask) | (mask & value);
+	ret = write_reg16(dev->config, REG_GPSR_LSB, GPSR);
+	if (ret == 0) {
+		drvdata->GPSR = GPSR;
+	}
+	k_sem_give(&drvdata->lock);
+
+	return ret;
+}
+
+static int stmpe1600_port_set_bits_raw(const struct device *dev, uint32_t mask)
+{
+	return stmpe1600_port_set_masked_raw(dev, mask, mask);
+}
+
+static int stmpe1600_port_clear_bits_raw(const struct device *dev, uint32_t mask)
+{
+	return stmpe1600_port_set_masked_raw(dev, mask, 0);
+}
+
+static int stmpe1600_port_toggle_bits(const struct device *dev, uint32_t mask)
+{
+	int ret;
+	uint16_t GPSR;
+	struct stmpe1600_drvdata *const drvdata = (struct stmpe1600_drvdata *const)dev->data;
+
+	if (k_is_in_isr()) {
+		return -EWOULDBLOCK;
+	}
+	k_sem_take(&drvdata->lock, K_FOREVER);
+	GPSR = drvdata->GPSR ^ mask;
+	ret = write_reg16(dev->config, REG_GPSR_LSB, GPSR);
+	if (ret == 0){
+		drvdata->GPSR = GPSR;
+	}
+	k_sem_give(&drvdata->lock);
+	return ret;
+}
+
+static int stmpe1600_pin_interrupt_configure(const struct device *dev,
+					     gpio_pin_t pin,
+					     enum gpio_int_mode mode,
+					     enum gpio_int_trig trig)
+{
+	return -ENOTSUP;
+}
+
+static int stmpe1600_init(const struct device *dev)
+{
+	int ret;
+	uint16_t chip_id;
+	const struct stmpe1600_config *const config = dev->config;
+	struct stmpe1600_drvdata *const drvdata = (struct stmpe1600_drvdata *const) dev->data;
```

Comment:
Looking at the source code of other gpio drivers we mostly start the variable definition block with getting pointers to `dev->config` and `dev->data` followed by remaining variable definitions and `ret` the last in the list. Let's write this block as
```
	const struct stmpe1600_config *const config = dev->config;
	struct stmpe1600_drvdata *const drvdata = (struct stmpe1600_drvdata *const) dev->data;
	uint16_t chip_id;
	int ret;
```
Here and in the remaining functions. Just to keep the style aligned with other drivers.